sycl/doc/extensions/LevelZeroBackend/LevelZeroBackend.md

```diff
@@ -150,6 +150,25 @@ struct {
 }
 ```
 </td>
+</tr><tr>
+<td>kernel</td>
+<td>
+
+``` C++
+ze_kernel_handle_t
+```
+</td>
+<td>
+
+``` C++
+struct {
+  kernel_bundle<bundle_state::executable> KernelBundle;
+  ze_kernel_handle_t NativeHandle;
+  ext::oneapi::level_zero::ownership Ownership{
+      ext::oneapi::level_zero::ownership::transfer};
+}
+```
+</td>

```

Comment:
Transferring an offline conversation here ...

I think we should not require the user to pass a `sycl::kernel_bundle` as a parameter to this call because most users will have only a `ze_module_handle_t`. Therefore, the common use case will require a call to `make_kernel_bundle()` like this:

```
ze_module_build_log_handle_t ZeBuildLog;
ze_module_handle_t ZeModule;
ze_result_t ZeResult =
  zeModuleCreate(ZeContext, ZeDevice, &ZeModuleDesc, &ZeModule, &ZeBuildLog);

kernel_bundle<bundle_state::executable> SyclKernelBundle =
  make_kernel_bundle<backend::ext_oneapi_level_zero, bundle_state::executable>({ZeModule}, Context);

ze_kernel_handle_t ZeKernel;
ze_kernel_desc_t ZeKernelDesc{ZE_STRUCTURE_TYPE_KERNEL_DESC, nullptr, 0, "my_kernel"};
ZeResult = zeKernelCreate(ZeModule, &ZeKernelDesc, &ZeKernel);

sycl::kernel SyclKernel =
  sycl::make_kernel<sycl::backend::ext_oneapi_level_zero>({SyclKernelBundle, ZeKernel}, Context);
```

I think it would be better to accept a `ze_module_handle_t` instead like this:

```
ze_module_build_log_handle_t ZeBuildLog;
ze_module_handle_t ZeModule;
ze_result_t ZeResult =
  zeModuleCreate(ZeContext, ZeDevice, &ZeModuleDesc, &ZeModule, &ZeBuildLog);

ze_kernel_handle_t ZeKernel;
ze_kernel_desc_t ZeKernelDesc{ZE_STRUCTURE_TYPE_KERNEL_DESC, nullptr, 0, "my_kernel"};
ZeResult = zeKernelCreate(ZeModule, &ZeKernelDesc, &ZeKernel);

sycl::kernel SyclKernel =
  sycl::make_kernel<sycl::backend::ext_oneapi_level_zero>({ZeModule, ZeKernel}, Context);
```

In the offline conversation, you expressed a concern that the user needs a way to control the ownership of the `ze_module_handle_t`. I suggested adding two "ownership" flags to the API, one for each handle. Since the flags have default values, most users won't specify either, therefore this won't add extra verbosity unless a user wants to "keep" ownership of the `ze_module_handle_t`, which seems like the less common case.

In your last comment, you said that you prefer a single "ownership" flag but did not elaborate.

Do you propose a single "ownership" flag that control the ownership of both handles?

Reply:
No, I mean each interop API to have single native handle and single ownership flag. Thus, one will have to "make_kernel_bundle" and then "make_kernel". Don't see any issue with verbosity here.

Reply:
Note also that this way a module handle only has its ownership specified once, in make_kernel_bundle, whereas otherwise it would be specified multiple times, additionally in each "make_kernel" call. Please approve if you don't have strong objection.

Reply:
Yes, that's a good point; you've convinced me.